src/Text/Pandoc/Writers/HTML.hs

```diff
@@ -933,9 +933,11 @@ inlineToHtml opts inline = do
                         linkText <- inlineListToHtml opts txt
                         slideVariant <- gets stSlideVariant
                         let s' = case s of
-                                      '#':xs | slideVariant == RevealJsSlides
-                                        -> '#':'/':xs
-                                      _ -> s
+                                   '#':xs -> let prefix = if slideVariant == RevealJsSlides
```

Comment:
This fix adds the prefix to all links starting with '#'.  This goes too far, I think.  The `--id-prefix` option as documented is only supposed to modify automatically generated identifiers.  So how do we know if the link is to an automatically generated identifier or a manually inserted one?